lib/resolver.js

```diff
@@ -49,37 +49,53 @@ class SchemaResolver {
 
     resolve(schema = this.root) {
         let resolvedSchema;
-        if (schema.type) {
-            resolvedSchema =this.resolveType(schema);
-        } else if (schema.anyOf) {
-            resolvedSchema = this.resolveAnyOf(schema);
-        } else if (schema.allOf) {
-            resolvedSchema = this.resolveAllOf(schema);
-        } else if (schema.oneOf) {
-            resolvedSchema = this.resolveOneOf(schema);
-        } else if (schema.not) {
-            resolvedSchema = this.resolveNot(schema);
-        } else if (schema.$ref) {
-            resolvedSchema = this.resolve(this.resolveReference(schema.$ref));
-        } else if (schema.enum) {
-            // If no type is specified, just enum
-            resolvedSchema = this.joi.any().valid(...schema.enum);
-        } else if (typeof schema === 'string') {
+
+        if (typeof schema === 'string') {
             // If schema is itself a string, interpret it as a type
             resolvedSchema = this.resolveType({ type: schema });
+        } else if (schema.$ref) {
+            resolvedSchema = this.resolve(this.resolveReference(schema.$ref));
         } else {
-            //Fall through to whatever.
-            //eslint-disable-next-line no-console
-            console.warn('WARNING: schema missing a \'type\' or \'$ref\' or \'enum\': \n%s', JSON.stringify(schema, null, 2));
-            //TODO: Handle better
-            resolvedSchema = this.joi.any();
+            const partialSchemas = [];
+            if (schema.type) {
+                partialSchemas.push(this.resolveType(schema));
+            } else if (schema.properties) {
+                // if no type is specified, just properties
+                partialSchemas.push(this.object(schema))
+            } else if (schema.format) {
```

Comment:
What if `schema.properties` and `schema.format` are both present?

Reply:
Seemse like these else-if statements could be under a single else case for code clarity.

Reply:
> What if `schema.properties` and `schema.format` are both present?

Then `schema.properties` will take precedence. We couldn't correctly handle them both at the same time without complex Joi conditional logic. As I mention in my commit messages, the handling of `properties` and `format` are stop-gap implementations. Handling them correctly would require *all* keywords on a schema to be handled, rather than just handling subsets of keywords based on the detected type. This structural shortcoming is captured here: https://gitlab.yum.com/yc-pos/poseidon-api.git/

Reply:
> Seemse like these else-if statements could be under a single else case for code clarity.

These first clauses are part of an if-else because that's generating the "base" schema. The base schema must allow the value to be _one of many types_ and then conditionally based on whichever type it might be, certain keywords are applied and others are ignored. This is why we currently use else/if here. The alternative is trying to implement that conditional matching against type in Joi conditionals.
 
The other are not part of the else since they generate additional schemas that we MUST match against, thus they are part of the .match('all').